Review request for the week final mobile first portfolio.

mercymugambi opened the pull request:
Hello Reviewer,
This is a Review request for the week final mobile first portfolio.

The project is based on figma template 1.
All concept learn this week has been implemented on this project.

mercymugambi committed:
Create html structure for projects section

mercymugambi committed:
Add project-section css

mercymugambi committed:
Add grid layout to skills section

mercymugambi committed:
Add button css

mercymugambi committed:
Add cards to index

mercymugambi committed:
Add arrow and separator to css

mercymugambi committed:
Check linter errors

mercymugambi committed:
Linter changes

mercymugambi committed:
Linter changes

mercymugambi committed:
Linter changes

mercymugambi committed:
Linter changes

didierganthier commented on index.html:
```diff
@@ -64,5 +66,182 @@ <h1>I’m MERCY <br />Glad to see <br />you!</h1>
       </div>
     </section>
+    <!--Start of detailed section-->
+    <section class="projects-section" >
+      <!--Card1-->
+      <div class="cards">
+        <img src="/assets/images/project1.png" alt="project1" />
+      </div>
+      <div class="cards-details">
+        <div>
+        <h2 class="title">Tonic</h2>
+        <div class="highlights">
+          <p class="highlight-text">Canopy</p>
+            <img class="counter" src="/assets/images/Counter.png" alt="Counter" />
+            <p class="countertext">Back End Dev</p>
+            <img src="/assets/images/Counter.png" alt="Counter" />
+            <p class="countertext">2015</p>
+        </div>
+        </div>
+        <p class="cards-description">
+          A daily selection of privately personalized reads; no accounts or
+          sign-ups required.
+        </p>
+        <div class="grid-skills">
+          <div class="item1">html</div>
+          <div class="item2">css</div>
+          <div class="item3">javascript</div>
+        </div>
+        <button id="popup">see project</button>
+      </div>
+    </section>
```
Well done on the `section` tag here 😉 but the requirement says that the container of the **Works** section should be a `section` tag. You indeed used a `section` tag but to wrap each work section, you're supposed to have a `section` tag wrapping all the other sections that you have for each work. This is what your structure looks like 👇🏽 
![Screenshot 2023-02-17 at 21 15 20](https://user-images.githubusercontent.com/45739963/219826495-ffe9fcb3-9b20-484c-928e-2bd5d3d201d0.png)

Also because of that, you're not fulfilling another requirement which is that the main container of the projects has a `display:grid`. This will help also when it's time to make your page responsive. 💯

didierganthier commented on index.html:
```diff
@@ -64,5 +66,182 @@ <h1>I’m MERCY <br />Glad to see <br />you!</h1>
+          A daily selection of privately personalized reads; no accounts or
+          sign-ups required.
+        </p>
+        <div class="grid-skills">
+          <div class="item1">html</div>
+          <div class="item2">css</div>
+          <div class="item3">javascript</div>
+        </div>
```
Well done here 💯  but to improve it even more the list of project technologies (e.g., "JavasScript", "HTML" and "CSS") should be a list `ul` and each of them should be a `li`. This will help browsers understand your page more. 😉

didierganthier commented on index.html:
```diff
@@ -64,5 +66,182 @@ <h1>I’m MERCY <br />Glad to see <br />you!</h1>
+    </section>
+    <!--About me-->
+    <section class="heading1" id="about">
+      <div class="header">
+        <h1>I’m MERCY</h1>
+      </div>
+      <div class="heading-paragraph">
+        <p>
+          Hello I’m a software developer! I can help you build a product , feature or website Look through some of my work and experience! If you like what you see and have a project you need coded, don’t hestiate to contact me.
+        </p>
+      </div>
```
Hey 👋🏽 please use the right content and the right design here. Your `About` section doesn't fit the Figma design. The main differences are the spacing and the rounded corner of the About section. See below 👇🏽 

## Figma
![Screenshot 2023-02-17 at 21 23 41](https://user-images.githubusercontent.com/45739963/219826859-a553ea03-3481-4735-8d6d-f638625271fe.png)

## Your design
![Screenshot 2023-02-17 at 21 24 13](https://user-images.githubusercontent.com/45739963/219826877-d72ba1c6-be28-4238-a934-5342f25733c0.png)

didierganthier commented on index.html:
```diff
@@ -64,5 +66,182 @@ <h1>I’m MERCY <br />Glad to see <br />you!</h1>
+        </ul>
+        <button id="popup" class="resume">Get my resume</button>
+      </div>
+      <div>
+        <p class="lang">Languages</p>
+        <img src="/assets/images/arrow-down.png" alt="img"class="arrow-down"> 
+      </div>
+      <div class="grid-languages">
+        <div class="item1" >
+          <img src="/assets/images/js.png" alt="js">html
+        </div>
+        <div class="item2"><img src="/assets/images/html.png" alt="html">css</div>
+        <div class="item3"><img src="/assets/images/css.png" alt="css">javascript</div>
+      </div>
+      <div class="lang">
+        <p>Frameworks</p>
+        <img src="/assets/images/arrow.png" alt="icon" class="arrow">
+        <img src="/assets/images/Separator Bottom.png" alt="Separator">
+        <p>Skills</p>
+        <img src="/assets/images/arrow.png" alt="icon" class="arrow2">
+        <img src="/assets/images/Separator Bottom.png" alt="Separator">
+      </div>
+      </div>
```
Looks cool 😎  but to make it even better the technology categories ("Languages", "Frameworks" and "Skills") should be in a `ul`, each of them in a `li`. This will help browsers understand that you're referencing a list.

didierganthier replied:
In order to make your design more aligned and improve the UI, these arrows should be aligned. Look at the differences 👇🏽 

## Figma
![Screenshot 2023-02-17 at 21 43 45](https://user-images.githubusercontent.com/45739963/219827650-8040e268-aeaa-4522-ac54-c65a52535a88.png)

## Your design
![Screenshot 2023-02-17 at 21 44 09](https://user-images.githubusercontent.com/45739963/219827675-4f44f909-1de9-44a1-9472-642ad49f6799.png)

didierganthier commented on index.html:
```diff
@@ -64,5 +66,182 @@ <h1>I’m MERCY <br />Glad to see <br />you!</h1>
+        <p class="lang">Languages</p>
+        <img src="/assets/images/arrow-down.png" alt="img"class="arrow-down"> 
+      </div>
+      <div class="grid-languages">
+        <div class="item1" >
+          <img src="/assets/images/js.png" alt="js">html
+        </div>
+        <div class="item2"><img src="/assets/images/html.png" alt="html">css</div>
+        <div class="item3"><img src="/assets/images/css.png" alt="css">javascript</div>
+      </div>
```
Great set of languages here 🤯 but to make it even better the technology in each category(e.g., "JavasScript", "HTML" and "CSS") should also be a list `ul` and each of them a `li`. This will help browsers understand that you're referencing a list.

didierganthier commented on style.css:
```diff
@@ -87,3 +88,212 @@ body {
   justify-content: flex-start;
 }
+
+.projects-section {
+  display: flex;
+  flex-direction: column;
+  padding: 16px;
+  gap: 12px;
+  margin-top: 650px;
+  width: 327px;
+  height: 552px;
+  background: #fff;
+  border: 1px solid #dfe1e6;
+  border-radius: 16px;
+}
+
+.cards {
+  width: 295px;
+  height: 220px;
+  border-radius: 8px;
+  flex: none;
+  order: 0;
+  flex-grow: 0;
+}
```
This is a nice card 👍🏽 but it's not centered yet, make sure to center it so that it matches the design. See the differences 👇🏽 

## Figma
![Screenshot 2023-02-17 at 21 29 31](https://user-images.githubusercontent.com/45739963/219827063-bf199e11-9301-4c34-8d50-60c5467d652f.png)

## Your design
![Screenshot 2023-02-17 at 21 29 54](https://user-images.githubusercontent.com/45739963/219827075-2cd6f05e-7e4f-4653-9c7f-a66e8b6af13b.png)

didierganthier commented on style.css:
```diff
@@ -87,3 +88,212 @@ body {
+  order: 1;
+  flex-grow: 0;
+}
+
+.highlight-text {
+  width: 55px;
+  height: 16px;
+  text-transform: uppercase;
+  font-family: "Poppins", sans-serif;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 13px;
+  line-height: 16px;
+  color: #344563;
+  flex: none;
+  order: 0;
+  flex-grow: 0;
+}
+
+.counter {
+  width: 8px;
+  height: 8px;
+  background: #c1c7d0;
+  flex: none;
+  flex-grow: 0;
+}
+
+.countertext {
+  width: 88px;
+  height: 16px;
+  font-family: "Poppins", sans-serif;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 13px;
+  line-height: 16px;
+  color: #7a869a;
+  flex: none;
+  order: 0;
+  flex-grow: 0;
+}
```
Great work 💯 but keep going, the main problem is the counter image, for now, it looks broken. See below 👇🏽

## Figma
![Screenshot 2023-02-17 at 21 32 57](https://user-images.githubusercontent.com/45739963/219827183-ca5bf679-e361-4226-ad0f-2e623db4bbcf.png)

## Your design
![Screenshot 2023-02-17 at 21 33 17](https://user-images.githubusercontent.com/45739963/219827198-36575754-6173-4012-bf3b-cc003c038fd3.png)

didierganthier commented on style.css:
```diff
@@ -87,3 +88,212 @@ body {
+  flex-grow: 0;
+}
+
+.grid-skills {
+  display: grid;
+  grid-template-columns: auto auto auto;
+  grid-gap: 10px;
+  padding: 10px;
+  align-items: center;
+  list-style: none;
+  margin: -20px 0 0 -10px;
+  gap: 8px;
+}
+
+.grid-skills > div {
+  text-align: center;
+  font-family: "Poppins", sans-serif;
+  font-style: normal;
+  font-size: 12px;
+  padding: 4px 12px;
+  width: 45px;
+  height: 24px;
+  font-weight: 500;
+  letter-spacing: 0.03rem;
+  line-height: 16px;
+  color: #6070ff;
+  background: #ebebff;
+  border-radius: 8px;
+}
```
Looks nice but I feel like the padding is overused, consider improving on that 😅 take a look at the differences below 👇🏽 

## Figma
![Screenshot 2023-02-17 at 21 35 40](https://user-images.githubusercontent.com/45739963/219827336-6ff85a59-08ac-43e1-9f9e-7eb762eae0af.png)



## Your design
![Screenshot 2023-02-17 at 21 34 24](https://user-images.githubusercontent.com/45739963/219827249-e5d075f8-a92d-4430-84f1-805cd31641fe.png)

Would look nicer, right? 😎

didierganthier commented on index.html:
```diff
@@ -64,5 +66,182 @@ <h1>I’m MERCY <br />Glad to see <br />you!</h1>
+        <img src="/assets/images/arrow-down.png" alt="img"class="arrow-down"> 
+      </div>
+      <div class="grid-languages">
+        <div class="item1" >
+          <img src="/assets/images/js.png" alt="js">html
+        </div>
+        <div class="item2"><img src="/assets/images/html.png" alt="html">css</div>
+        <div class="item3"><img src="/assets/images/css.png" alt="css">javascript</div>
```
This is great but it could be improved with some spacing and by centering the image with the text. See below 👇🏽 

## Figma
![Screenshot 2023-02-17 at 21 40 57](https://user-images.githubusercontent.com/45739963/219827554-bbc1ae6e-8c76-41f4-ba4a-48a0c16796dd.png)


## Your design
![Screenshot 2023-02-17 at 21 41 19](https://user-images.githubusercontent.com/45739963/219827566-684e8273-a09c-48a8-a34b-53148e0ef3eb.png)

didierganthier commented on style.css:
```diff
@@ -87,3 +88,212 @@ body {
+  border-radius: 8px;
+}
+
+#popup {
+  box-sizing: border-box;
+  border: 1px solid #7f8cff;
+  background-color: #fff;
+  letter-spacing: 0.03rem;
+  line-height: 24px;
+  font-family: "Poppins", sans-serif;
+  border-radius: 8px;
+  padding: 12px;
+  bottom: 10px;
+  gap: 10px;
+  cursor: pointer;
+  text-align: center;
+  font-size: 17px;
+  font-weight: 500;
+  font-style: normal;
+  color: #396df2;
+}
```
This is a nice button, I love it 😄 but it could look/behave even better and offer a better UX/UI. Yours looks too large for now. See the differences 👇🏽 

## Figma 
![Screenshot 2023-02-17 at 20 50 12](https://user-images.githubusercontent.com/45739963/219825349-d5258193-9de9-4a2d-9c48-3e917d1770fb.png)

## Your design
![Screenshot 2023-02-17 at 21 46 04](https://user-images.githubusercontent.com/45739963/219827746-281e6bbf-d1a3-4aa0-9219-9c8a83e98098.png)

Also, make sure the button (also the `Get resume` button) follows the button styling: Active, Hover, and Pressed (see below 👇🏽 )

![Screenshot 2023-02-17 at 20 52 56](https://user-images.githubusercontent.com/45739963/219825480-90aea75f-5322-4e74-b6c2-5a8ebc18523b.png)

You can accomplish that by adding the `#popup:hover` style 💯

didierganthier reviewed (changes requested):
Hi @mercymugambi ,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.

<img src="https://media.giphy.com/media/3oEduMcvxTNPlGuhH2/giphy.gif" />

## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct GitHub flow ✔️ 

## Required Changes 🛠
- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code.

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

mercymugambi committed:
Delete details section for modification

mercymugambi committed:
Updated html content into one section

mercymugambi committed:
Structure about section as required

mercymugambi committed:
Add CSS to about section

mercymugambi committed:
Add hover to the buttons

mercymugambi committed:
Remove fontAwesome link

mercymugambi commented:
Hello Reviewer,
I have managed to amend all the changes as requested. These changes are as below:

- used single section tag for all cards.
- linked projects section to li tags.
- Used the figma content on the about section.
- Technology category, i used un, li tags for better browser referencing.
- counter image from my end looks fine after testing (not broken). Though did changes to be accomodated on other browsers
- Removed the over used padding on skills section.
- Spacing updated on skills section.
- Arrows aligned well.
- Added the button styling on hover and whe active. 
Thank you!

mercymugambi committed:
Fix div to accomodate button

RaoAkif commented on index.html:
```diff
@@ -8,60 +8,246 @@
+      </div>
+    </section>
+    <!--about-->
+    <section class="about">
+      <div class="header1">
+        <h2 class="header">
+          About<br />
+          Myself
+        </h2>
+        <article class="heading-paragraph">
+          <p>
+            Hello I’m a software developer! I can help you build a product ,
+            feature or website Look through some of my work and experience! If
+            you like what you see and have a project you need coded, don’t
+            hestiate to contact me.
+          </p>
+        </article>
+        <div class="heading-social">
+        <h2 class="icons-header">LET'S CONNECT</h2>
+        <div>
```
As suggested by the previous reviewer
`Please use the right content and the right design here. Your About section doesn't fit the Figma design. The main differences are the spacing and the rounded corner of the About section. See below 👇🏽`

![image](https://user-images.githubusercontent.com/45739963/219826859-a553ea03-3481-4735-8d6d-f638625271fe.png)

RaoAkif commented on index.html:
```diff
@@ -8,60 +8,246 @@
+        <button type="button" id="popup">Get my resume</button>
+      </div>
+      </div>
+      <div id="about">
+        <ul class="lang">
+          <li><h4>Languages</h4></li>
+          <li><img src="./assets/images/arrow-down.png" alt="arrow.png" /></li>
+        </ul>
+        <ul class="icons-lang">
+          <li><img src="./assets/images/js.png" alt="js" />Javascript</li>
+          <li><img src="./assets/images/html.png" alt="html" />HTML</li>
+          <li><img src="./assets/images/css.png" alt="Ellipse" />CSS</li>
+        </ul>
+        <ul class="lang">
```
As suggested by the previous reviewer, Kindly push your arrow a little bit left as shown in the comparison below
| Your Design   | FIGMA Design  |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/61361037/219875578-cd73a93d-2ec0-4ad1-90db-efa854b0183b.png)| ![image](https://user-images.githubusercontent.com/61361037/219875559-ee48afb7-45b1-4620-9f7b-c329a903cab3.png)|

RaoAkif reviewed (changes requested):
## STATUS: REQUIRED CHANGES ♻️

Hi @mercymugambi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights
✔ Linters are on point 
✔ Splendid work with the README file
✔ Great use of GitHub flow
✔ Good code design and implementation

## Required Changes ♻️
_Check the comments under the review._
<img src="https://camo.githubusercontent.com/6ac905aa3a075ef45de55758c20c6e6efb1dfdbbb0d9fd03d4718d26c5c96cde/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f4a713832345239334a734c775a436169534c2f67697068792e676966" />

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mercymugambi committed:
Add spacing and border radius to about

mercymugambi committed:
Push arrows to left

mercymugambi committed:
Put lang container into div tag

mercymugambi commented:
Hi,
The changes requested on implementing spaces and rounded corner (border-top-right-radius: 50px) are done.
suggestion to push arrow to left is done too as shown. Please review and consider
![arrows](https://user-images.githubusercontent.com/48310207/219879776-c0565c26-df04-47c3-a1da-9bae0ef6010f.PNG)

jerryowusu reviewed (approved):
Hi @mercymugambi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![200w](https://user-images.githubusercontent.com/65233860/219880389-f09d92cb-deac-4db6-a6d4-82e492db78f8.gif)


## Highlights
: No linter errors :+1: 
Great effort so far :+1: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
